evadb/executor/sqlalchemy_executor.py

```diff
@@ -0,0 +1,54 @@
+# coding=utf-8
+# Copyright 2018-2023 EvaDB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Iterator
+
+import pandas as pd
+from sqlalchemy import create_engine
+
+from evadb.catalog.catalog_utils import generate_sqlalchemy_conn_str
+from evadb.database import EvaDBDatabase
+from evadb.executor.abstract_executor import AbstractExecutor
+from evadb.models.storage.batch import Batch
+from evadb.plan_nodes.native_plan import SQLAlchemyPlan
+
+
+class SQLAlchemyExecutor(AbstractExecutor):
+    """
+    Execute SQL through SQLAlchemy engine.
+    """
+
+    def __init__(self, db: EvaDBDatabase, node: SQLAlchemyPlan):
+        super().__init__(db, node)
+        self._database_name = node.database_name
+        self._query_string = node.query_string
+
+    def exec(self, *args, **kwargs) -> Iterator[Batch]:
+        db_catalog_entry = self.db.catalog().get_database_catalog_entry(
+            self._database_name
+        )
+
+        conn_str = generate_sqlalchemy_conn_str(
+            db_catalog_entry.engine,
+            db_catalog_entry.params,
+        )
+
+        engine = create_engine(conn_str)
+
+        with engine.connect() as con:
+            if "SELECT" in self._query_string or "select" in self._query_string:
```

Comment:
We need to make this robust for cases where select is used for creating tables etc. Also, if query contains `Select` etc

Reply:
What is the output for `CREATE TABLE` from sqlalchemy? Is it possible to simply execute the query and yield whatever output sqlalchemy yields in dataframe type?

Reply:
I like this idea

Reply:
How about we replicate this? https://github.com/mindsdb/mindsdb/blame/b417ecff78cca2e7f09d4e7daa3461595c77ad0d/mindsdb/integrations/handlers/postgres_handler/postgres_handler.py

I'm using it in the other PR for testing the connection.

Reply:
https://github.com/sqlalchemy/sqlalchemy/issues/5433#issuecomment-652705643

Based on what I see, it is still true for this API for latest SQLAlchemy. There is no rows returned besides `SELECT`.